go/vt/wrangler/vexec.go

```diff
@@ -293,25 +328,30 @@ func (wr *Wrangler) convertQueryResultToSQLTypesResult(results map[*topo.TabletI
 	return retResults
 }
 
-// WorkflowAction can start/stop/delete or list streams in _vt.vreplication on all primaries in the target keyspace of the workflow.
-func (wr *Wrangler) WorkflowAction(ctx context.Context, workflow, keyspace, action string, dryRun bool) (map[*topo.TabletInfo]*sqltypes.Result, error) {
-
-	if action == "show" {
+// WorkflowAction can start/stop/update/delete or list streams in _vt.vreplication
+// on all primaries in the target keyspace of the workflow.
+// rpcReq is an optional argument for any actions that use the new RPC path. Today
+// that is only the update action. When using the SQL interface this is ignored and
+// you can pass nil.
```

Comment:
is the plan to eventually move off of `any` once all the subcommands have been moved? it feels kinda bad that we have to go from typed=>`any` and immediately back to typed "just" to get some code sharing

Reply:
I believe that we still have to keep this code around long-term, e.g. for the stand-alone `vtctl` client, so the plan is to move all of the workflow actions (and any other commands/actions using the raw SQL interface) to RPCs and they'll each have a different request type. So I was thinking long term here.

This is also very much not in any sort of hot path, so a good use of generics IMO (I virtually never use them).

Reply:
this isn't "generics" though, this is just `s/interface{}/any`, so we still have to do the type assertion at the end rather than getting the compiler-monomorphisation. i'd rather see us end up (not necessarily now, but in ~~ the fuuuuuuuture ~~) where either

1. we have shared behavior that is actually generic over some interface that all the different sub-command requests satisfy _or_
2. we refactor this so the common bits don't need to take a request structure (or maybe they take a `validate` func, or a `callback`, or something, to dependency-inject the per-sub-command logic) and remove the `any`

Reply:
i'm not saying we need to fix this now, but i think an end state that does, roughly:

```go
// grpcvtctldserver
req := &vtctldatapb.WorkflowUpdateRequest{...}
wr.WorkflowAction(..., req)

// wrangler
func WorkflowAction(..., req any) {
    req.(*vtctldatapb.WorkflowUpdateRequest)
}
```

is less than ideal and we should avoid it, or think about how we can avoid it in the end

Reply:
I agree it's not ideal. I'm just trying to blend things during the transition from SQL to RPCs. It's a bit awkward while that's ongoing. I'm OK with it as I think the entire path should be refactored once they've all be migrated to RPCs.

Reply:
sounds good to me then! (i know migrations like this - where an "all at once" is way too big to manage (either for the author or reviewers), and "in pieces" has these sorts of tradeoffs - can be tricky!)